migrator/replica/db_replica_manager_impl.go

```diff
@@ -0,0 +1,117 @@
+		dbmRocks:             rocksdb.New(basePath, forceVersion),
+		dbmPostgres:          postgres.New(forceVersion, adminConfig, sourceMap),
+	}
+}
+
+// New - returns a new ready-to-use store.
+func New(basePath string, forceVersion string) *DBReplicaManagerImpl {
+	return &DBReplicaManagerImpl{
+		forceRollbackVersion: forceVersion,
+		basePath:             basePath,
+		dbmRocks:             rocksdb.New(basePath, forceVersion),
+	}
+}
+
+// Scan - checks the persistent data of central and gather the replica information
+// from the supported databases.
+func (d *DBReplicaManagerImpl) Scan() error {
+	err := d.dbmRocks.Scan()
+	if err != nil {
+		// If our focus is Postgres, just log the error and ignore Rocks
+		if features.PostgresDatastore.Enabled() {
+			log.Warn(err)
+		} else {
+			return err
+		}
+	}
+
+	if features.PostgresDatastore.Enabled() {
+		err = d.dbmPostgres.Scan()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// GetReplicaToMigrate - finds a replica to migrate.
+// It returns the replica link, path to database, postgres database name and error if fails.
+func (d *DBReplicaManagerImpl) GetReplicaToMigrate() (string, string, string, error) {
+	// Get the RocksDB replica we are migrating
+	replica, replicaPath, err := d.dbmRocks.GetReplicaToMigrate()
+	if err != nil {
+		return "", "", "", err
+	}
+
+	if features.PostgresDatastore.Enabled() {
+		// Get the version of the Rocks Current so Postgres manager can use that info
+		// to determine what replica it needs to migrate.
+		var rocksVersion *migrations.MigrationVersion
+		// If we found no replica to migrate from Rocks no need to check its version.  That means the
+		// mount does not exist
+		if replica != "" && replicaPath != "" {
+			rocksVersion = d.dbmRocks.GetVersion(rocksdb.CurrentReplica)
+			rocksVersion.LastPersisted = d.dbmRocks.GetCurrentReplicaCreationTime()
+		}
+
+		pgReplica, migrateFromRocks, err := d.dbmPostgres.GetReplicaToMigrate(rocksVersion)
+		if err != nil {
+			return "", "", "", err
+		}
+
+		// If we need to migrate from rocks we need to return both
+		// rocks and postgres replicas.  Otherwise we just need one or
+		// the other.
+		if migrateFromRocks {
+			return replica, replicaPath, pgReplica, err
+		}
+		return "", "", pgReplica, err
+
+	}
+
+	return replica, replicaPath, "", err
+}
+
+// Persist - replaces current replica with upgraded one.
+func (d *DBReplicaManagerImpl) Persist(replicaName string, pgReplica string, persistBoth bool) error {
+	if features.PostgresDatastore.Enabled() {
+		// We need to persist the Rocks previous, so it is there in case of a rollback.  In the case of
+		// an upgrade that will generate a previous, the Temp Replica will be the one RocksDB persists.
+		// During the persist operation the Current replica will move to Previous and Temp will move to Current.
+		if persistBoth && replicaName == rocksdb.TempReplica {
+			if err := d.dbmRocks.Persist(replicaName); err != nil {
```

Comment:
We did not remove the previous and current version after two migrations, do we? 

Like (rocksdb ver) -> (Postgres ver1) -> (postgres ver2). After two upgrades, we should be able to clean up all the replicas on rocksdb.

Reply:
Agreed.  I had looked at that a while ago and there are a lot of start up checks to ensure that current rocks exists.  I'd rather not delve into that in this PR because it is big enough.  So I wrote [ROX-12289](https://issues.redhat.com/browse/ROX-12289) to solve this issue.